code/modules/library/skill_learning/skillchip.dm

```diff
@@ -0,0 +1,84 @@
+
+	icon = 'icons/obj/card.dmi'
+	icon_state = "data_3"
+	custom_price = 500
+
+	/// Trait automatically granted by this chip, optional
+	var/auto_trait
+
+	/// Skill name shown on UI
+	var/skill_name
+
+	/// Skill description shown on UI
+	var/skill_description
+
+	/// FS icon show on UI
+	var/skill_icon = "brain"
+
+	/// Message shown when implanting the chip
+	var/implanting_message
+	/// Message shown when extracting the chip
+	var/removal_message
+
+	//If set to TRUE, trying to extract the chip will destroy it instead
+	var/removable = TRUE
+
+/// Called after implantation and/or brain entering new body
+/obj/item/skillchip/proc/on_apply(mob/living/carbon/user,silent=TRUE)
+	if(!silent && implanting_message)
+		to_chat(user,implanting_message)
+	if(auto_trait)
+		ADD_TRAIT(user,auto_trait,SKILLCHIP_TRAIT)
+
+/// Called after removal and/or brain exiting the body
+/obj/item/skillchip/proc/on_removal(mob/living/carbon/user,silent=TRUE)
+	if(!silent && removal_message)
+		to_chat(user,removal_message)
+	if(auto_trait)
+		REMOVE_TRAIT(user,auto_trait,SKILLCHIP_TRAIT)
+
+/// Checks if this implant is valid to implant in a given mob.
+/obj/item/skillchip/proc/can_be_implanted(mob/living/carbon/target)
+	//No brain
+	var/obj/item/organ/brain/B = target.getorganslot(ORGAN_SLOT_BRAIN)
+	if(!B)
+		return FALSE
+	//No skill slots left
+	if(length(B.skillchips) >= target.get_max_skillchip_count())
+		return FALSE
+	//Only one copy of each for now.
+	if(locate(type) in B.skillchips)
+		return FALSE
+	return TRUE
+
+/obj/item/skillchip/basketweaving
+	name = "Basketsoft 3000 skillchip"
+	desc = "Underwater edition."
+	auto_trait = TRAIT_UNDERWATER_BASKETWEAVING_KNOWLEDGE
+	skill_name = "Underwater Basketweaving"
+	skill_description = "Master intricate art of using twine to create perfect baskets while submerged."
+	skill_icon = "shopping-basket"
+	implanting_message = "<span class='notice'>You're one with the twine and the sea.</span>"
+	removal_message = "<span class='notice'>Higher mysteries of underwater basketweaving leave your mind.</span>"
+
+/obj/item/skillchip/wine_taster
+	name = "WINE skillchip"
+	desc = "Wine.Is.Not.Equal version 5."
+	auto_trait = TRAIT_WINE_TASTER
+	skill_name = "Wine Tasting"
+	skill_description = "Recognize wine vintage from taste alone. Never again lack an opinion when presented with an unknown drink."
+	skill_icon = "wine-bottle"
+	implanting_message = "<span class='notice'>You recall wine taste.</span>"
+	removal_message = "<span class='notice'>Your memories of wine evaporate.</span>"
+
+/obj/item/skillchip/bonsai
+	name = "Hedge 3 skillchip"
+	auto_trait = TRAIT_BONSAI
+	skill_name = "Hedgetrimming"
+	skill_description = "Trim hedges and potted plants into marvelous new shapes with any old knife. Not applicable to plastic plants."
+	skill_icon = "spa"
+	implanting_message = "<span class='notice'>Your minds is filled with plant arrangments.</span>"
```

Comment:
damn bro I didn't know that skill chips gave you extra minds

Reply:
shit actually you should add a split personality skillchip, that'd be cool

maybe you can even use that special split personality subtype that only triggers a switch upon hearing a special key word

Reply:
Fukken syndicate sleeper agent antag. Hears codeword and gets activated.